client/router.js

```diff
@@ -369,3 +427,56 @@ Router.prototype._updateCallbacks = function () {
 
 Router.prototype._page = page;
 Router.prototype._qs = qs;
+
+function shouldCallTrigger(current, fn) {
```

Comment:
You can create these as methods in the Router class. Then we can test them individually.